Ng-374: pop-up window when user is deleted from project

IamBigBobby opened the pull request:
task: https://github.com/epam/Indigo-ELN-v.-2.0/issues/374
screenshoot:  
<img width="959" height="497" alt="image" src="https://github.com/user-attachments/assets/bd1d3fc6-4d9e-487f-9895-576368da2113" />  
<img width="956" height="493" alt="image" src="https://github.com/user-attachments/assets/9e81d601-ae9d-4aee-aec8-7b2c99094492" />  
**IMPORTANT INFO**  
there is no feature, which removes member from all entities associated with project
need to get this endpoint (i leaved the TODO for this case in code)

serhatkaya commented on indigo-frontend/src/core/components/common/dropdown-menu/dropdown-menu.component.ts:
```diff
@@ -108,6 +109,12 @@ export class DropdownMenuComponent
 
     const newValue = item.value || item.label;
 
+    if (item.label.toLowerCase() === 'none' || item.value?.toLowerCase() === 'none') {
+      this.noneClicked.emit();
+      this.isOpen = false;
+      return;
+    }
+
```
You have implemented custom behavior in a generic component. You shouldn’t do that, as generic components are expected to remain behavior-agnostic.

serhatkaya commented on indigo-frontend/src/core/components/common/team/team.component.html:
```diff
@@ -75,10 +75,53 @@ <h2 class="font-semibold text-base">{{ config.title }}</h2>
+  #removeConfirmationModal
+  title="Remove Team Member"
+  [closeOnBackdropClick]="true"
+  [closeOnEsc]="true"
+>
+  <div class="space-y-4">
+    <p class="text-neutral-800">
+      Are you sure you want to remove
+      <strong>{{ memberToRemove?.displayName }}</strong>
+      from the team?
+    </p>
+    @if (hasNestedAccess) {
+      <label class="flex items-center gap-2 cursor-pointer">
+        <input
+          type="checkbox"
+          [(ngModel)]="removeFromAllEntities"
+          class="w-4 h-4 text-primary-400 border-neutral-300 rounded focus:ring-primary-400"
+        />
+        <span class="text-sm text-neutral-800">
+          Also, remove the user from all projects, notebooks and experiments
+        </span>
+      </label>
+    }
+  </div>
+
+  <ng-template #modalFooter>
+    <div class="flex justify-end gap-3">
+      <eln-button variant="grey-outline" (click)="cancelRemoveMember()">
+        Cancel
+      </eln-button>
+      <eln-button
+        variant="red"
+        (click)="confirmRemoveMember()"
+        [disabled]="loading().updatingMembers.size > 0"
+      >
+        Remove
+      </eln-button>
+    </div>
+  </ng-template>
+</eln-modal>
```
Consider creating a separate component for this and use MatDialog.

serhatkaya commented on indigo-frontend/src/core/components/common/team/team.component.ts:
```diff
@@ -152,6 +159,96 @@ export class TeamComponent implements OnInit {
+            map(nestedAccesses => {
+                const hasAccess = nestedAccesses.some(access => access.userId === userId);
+                return hasAccess;
+            })
```
can be simplified 

`map(nestedAccesses => nestedAccesses.some(na => na.userId === userId))`

serhatkaya commented on indigo-frontend/src/core/components/common/team/team.component.ts:
```diff
@@ -152,6 +159,96 @@ export class TeamComponent implements OnInit {
+            return new Observable<boolean>(observer => {
+                observer.next(false);
+                observer.complete();
+            });
```
you could use `of` function


return of(false);

serhatkaya commented on indigo-frontend/src/core/components/common/team/team.component.ts:
```diff
@@ -50,6 +54,9 @@ export class TeamComponent implements OnInit {
     selectedUserIds: string[] = [];
+    memberToRemove: ProjectAcl | null = null;
+    removeFromAllEntities = false;
+    hasNestedAccess = false;
```
You shouldn't create global variables for member-scoped value

aamialkovich reviewed (changes requested):
Need to make changes per Serhat's review.

Commit:
fix: resolve conflicts

feat: upd dropdown-menu.component and team.component.html according popup requirements

refactor: remove comments

feat: implement base for featur which remove member from all entities

feat: implement second version of popup logic

fix: fix template from 3.0 branch

Revert "feat: implement second version of popup logic"

This reverts commit 043d4b1d071bec9295e430ab78509764d16ebf9d.

feat: implement nested access endpoint

refactor: remove logs from checkNestedAccess

feat: implement deleting from all entities

refactor: remove logs

refactor: remove comments from onNoneClicked

feat: add error handling for confirmRemoveMember method

IamBigBobby commented:
there are a lot of changes in 3.0, which was after my pr opening, i tried to rebase and adapt my branch, but there are a lot of changes, which could affect on 3.0, and which i dont understand yet

so i made a decision to close my previous pr and do it again with fresh thoughts
